### PR TITLE
74 UI for create template view lee

### PR DESCRIPTION
**Pull request by UiHyun-Lee**
## 1. functional consistency
- [x] Does the code fulfill the specified requirements?
- [x] Have all requirements been considered?
- [x] Are there scenarios that the code may not handle?
- [ ] Could additional features be helpful?

## 2. code readability and structure
- [x] Is the code clear and easy to understand?
  - [ ] Structured indentation
  - [ ] Consistent naming
- [x] Are there relevant comments?
  - [ ] Especially for complex sections
- [x] Does the structure make the code easier to navigate?

## 3. performance aspects
- [ ] Are there possible bottlenecks in the code?
- [ ] Are there unnecessary repetitions or inefficient loops?
- [ ] Are resource-intensive operations optimized?
- [ ] Is the overall resource utilization efficient?

## 4. error handling and logging
- [x] Are possible error cases handled robustly?
- [x] Are the error messages clear and helpful?
- [x] Are there appropriate logging mechanisms for debugging and monitoring?

## 5. security aspects
- [ ] Could security vulnerabilities (e.g. SQL injection, DoS attacks) occur?
- [ ] Is data processing carried out securely and appropriately?
- [ ] Are there risks of data leaks or other security compromises?
- [ ] Were known security best practices followed?

## 6. tests and coverage
- [ ] Do tests exist for new functions or changes?
  - [ ] Unit tests
  - [ ] Integration tests
- [ ] Have existing tests been adapted accordingly?
- [ ] Are edge cases considered?
- [ ] Is the test coverage sufficient?

## 7. code duplication
- [ ] Is there repeated or redundant code?
- [ ] Can existing functions or libraries be used?
- [ ] Is readability and maintainability improved by avoiding duplicates?

## 8 Clarity and dependencies
- [ ] Are proven frameworks, libraries or components used?
- [ ] Are the dependencies up to date?
- [ ] Have obsolete or redundant dependencies been removed?
- [ ] Are the dependencies reliable and well maintained?

## 9. documentation and comments
- [ ] Are comments and documentation comprehensive and understandable?
  - [ ] High-level overviews
  - [ ] Inline comments for complex sections
  - [ ] Docstrings for functions and classes
- [ ] Is the documentation kept consistent and up-to-date?
- [ ] Does the documentation make work easier for future developers?

## 10. Compliance with coding standards
- [ ] Does the code comply with the defined standards?
- [ ] Are the standards consistently adhered to?
- [ ] Does compliance contribute to maintainability and error reduction?
- [ ] Are the standards clearly documented and understandable for the team?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -244,7 +245,11 @@
                     <execution>
                         <goals>
                             <goal>prepare-frontend</goal>
+                            <goal>build-frontend</goal>
                         </goals>
+                        <configuration>
+                            <optimizeBundle>false</optimizeBundle>
```
Comment (WernersGit) [inline pom.xml:251]: We also need to determine the effect of the optimizeBundle configuration on all other components

File: Dockerfile [hunk 2]
```diff
@@ -32,6 +32,6 @@ COPY target/*.jar app.jar
 EXPOSE 8080
 
 # Run the trust-cert script on container start and then start the Java application
-ENTRYPOINT ["/bin/bash", "-c", "/usr/local/bin/install_trust_certs.sh && java -jar /app.jar"]
+#ENTRYPOINT ["/bin/bash", "-c", "/usr/local/bin/install_trust_certs.sh && java -jar /app.jar"]
 # ENTRYPOINT ["/bin/bash", "-c", "/usr/local/bin/install_trust_certs.sh && java -Djavax.net.ssl.trustStore=/opt/java/openjdk/lib/security/cacerts -Djavax.net.ssl.trustStorePassword=changeit -Djdk.tls.client.protocols=TLSv1.3 -jar /app.jar"]
-# ENTRYPOINT ["/bin/bash", "-c", "/usr/local/bin/install_trust_certs.sh && java -Dhttp.proxyHost=host.docker.internal -Dhttp.proxyPort=8088 -Dhttps.proxyHost=host.docker.internal -Dhttps.proxyPort=8088 -jar /app.jar"]
+ ENTRYPOINT ["/bin/bash", "-c", "/usr/local/bin/install_trust_certs.sh && java -Dhttp.proxyHost=host.docker.internal -Dhttp.proxyPort=8088 -Dhttps.proxyHost=host.docker.internal -Dhttps.proxyPort=8088 -jar /app.jar"]
```
Comment (WernersGit) [inline Dockerfile:37]: We must not push this configuration into the main branch

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/CreateTemplate.java [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
-import com.vaadin.flow.theme.lumo.LumoUtility.Gap;
-import com.vaadin.flow.theme.lumo.LumoUtility.Padding;
-import java.util.ArrayList;
-import java.util.List;
+import com.vaadin.flow.spring.annotation.UIScope;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
 
 /**
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/CreateTemplate.java:21]: java doc is incomplete

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/CreateTemplate.java [hunk 4]
```diff
@@ -24,113 +25,103 @@
...
-    private void setSelectSampleData(Select select) {
-        List<SampleItem> sampleItems = new ArrayList<>();
-        sampleItems.add(new SampleItem("first", "v1", null));
-        sampleItems.add(new SampleItem("second", "v2", null));
-        sampleItems.add(new SampleItem("third", "v3", null));
-        sampleItems.add(new SampleItem("fourth", "v4", null));
-        select.setItems(sampleItems);
-        select.setItemLabelGenerator(item -> ((SampleItem) item).label());
-        select.setItemEnabledProvider(item -> !Boolean.TRUE.equals(((SampleItem) item).disabled()));
+    private void setParserDropdownItems(Select<ParserType> parserDropdown) {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/CreateTemplate.java:85]: no java doc

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/CreateTemplate.java [hunk 5]
```diff
@@ -24,113 +25,103 @@
...
+        parserDropdown.setItems(
+                Arrays.stream(ParserType.values())
+                        .filter(type -> !type.name().startsWith("_"))
+                        .collect(Collectors.toList())
+        );
     }
+    private void updateUIForSelection(ParserType selectedValue) {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/CreateTemplate.java:98]: no java doc

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/Csv.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.component.orderedlayout.FlexComponent;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.router.PageTitle;
+import com.vaadin.flow.router.Route;
+
+@PageTitle("CSV")
+@Route("csv")
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/Csv.java:20]: allgemeine Route für alle Views (in Tabs zusammengefasst) /templates

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/Csv.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        skipLineMetadata.clear();
+        skipLineData.clear();
+        totalColumnsMetadata.clear();
+        totalColumnsData.clear();
+        hasHeadlineMetadata.clear();
+        hasHeadlineData.clear();
+        delimiterMetadata.clear();
+        delimiterData.clear();
+    }
+    private HorizontalLayout createAlignedRowLayout(String label, String infoLabel,TextField metadataField, TextField dataField) {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/Csv.java]: no java doc

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/Csv.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        infoIcon.getStyle().set("margin-right", "10px");
+
+        metadataField.setWidth("250px");
+        dataField.setWidth("250px");
+
+        layout.add(labelSpan, infoIcon, metadataField, dataField);
+        return layout;
+    }
+
+    private Icon createInfoIcon(String label) {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/Csv.java:118]: no java doc

**Review by SergentKim — APPROVED** [review]

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/SubCsv.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    private final TextField skipLineMetadata = new TextField();
+    private final TextField skipLineData = new TextField();
+    private final TextField totalColumnsMetadata = new TextField();
+    private final TextField totalColumnsData = new TextField();
+    private final TextField hasHeadlineMetadata = new TextField();
+    private final TextField hasHeadlineData = new TextField();
+    private final TextField delimiterMetadata = new TextField();
+    private final TextField delimiterData = new TextField();
+
+    public SubCsv(String string) {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/SubCsv.java:38]: no java doc

**Review by SergentKim — APPROVED** [review]

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/SubCsv.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+                createAlignedRowLayout("Total Columns", "Total amount of columns in the data.",totalColumnsMetadata),
+                createAlignedRowLayout("Delimiter", "Delimiter for data e.g. a csv delimiter.",delimiterMetadata),
+                createAlignedRowLayout("Skip Lines", "Amount of lines to skip after the headline (pattern).",skipLineMetadata)
+        );
+
+        getContent().setWidth("100%");
+        getContent().getStyle().set("flex-grow", "1");
+        getContent().add(verticalLayout);
+    }
+    public void clearAllFields() {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/SubCsv.java:67]: no java doc

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/SubCsv.java [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+        skipLineMetadata.clear();
+        skipLineData.clear();
+        totalColumnsMetadata.clear();
+        totalColumnsData.clear();
+        hasHeadlineMetadata.clear();
+        hasHeadlineData.clear();
+        delimiterMetadata.clear();
+        delimiterData.clear();
+    }
+    private HorizontalLayout createAlignedRowLayout(String label, String infoLabel,TextField metadataField) {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/SubCsv.java]: no java doc

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/SubCsv.java [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+        Icon infoIcon = createInfoIcon(infoLabel);
+        infoIcon.getStyle().set("margin-right", "10px");
+
+        metadataField.setWidth("250px");
+
+        layout.add(labelSpan, infoIcon, metadataField);
+        return layout;
+    }
+
+    private Icon createInfoIcon(String label) {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/csv/SubCsv.java:110]: no java doc

**Review by SergentKim — APPROVED** [review]

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/custom/Custom.java [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+                subTextData,
+                subCsvData
+        );
+
+        getContent().setWidth("100%");
+        getContent().getStyle().set("flex-grow", "1");
+        getContent().add(verticalLayout);
+    }
+
+    private void updateVisibility(ParserType selectedValue, String type) {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/custom/Custom.java:115]: no java doc

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/custom/Custom.java [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+import com.vaadin.flow.router.PageTitle;
+import com.vaadin.flow.router.Route;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@PageTitle("Custom")
+@Route("custom")
+public class Custom extends Composite<VerticalLayout> {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/custom/Custom.java:27]: @Route: content should be combined within one page using tabs: /template

Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/custom/Custom.java:27]: no java doc

File: src/main/java/com/devteam45ldm/ldm/views/createtemplate/custom/Custom.java [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+                    subJsonXmlData.setVisible(false);
+                    subCsvData.setVisible(false);
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+    public void clearAllFields() {
```
Comment (WernersGit) [inline src/main/java/com/devteam45ldm/ldm/views/createtemplate/custom/Custom.java:162]: no java doc

**Review by WernersGit — CHANGES_REQUESTED** [review]
Each method should contain a Java DOC (there are many more uncommented methods than I have marked).

**Comment by UiHyun-Lee** [comment]
java doc completed

**Review by WernersGit — APPROVED** [review]
ok
